src/invidious/partial_json_response.cr

```diff
@@ -0,0 +1,160 @@
+
+      property_nests.each do |nest|
+        nest_list = nest.split('/')
+        property_value = get_hash_property(hash, nest_list)
+        set_hash_property(final_hash, nest_list, property_value)
+      end
+    end
+
+    # next, handle nest groups: items(id, etag, etc)
+    regex_index = 0
+    text_index = 0
+    while regex_result = self.nest_group_pattern.match(fields_text, regex_index)
+      raw_match = regex_result[0]
+      group_name = regex_result["groupname"]
+
+      regex_index = regex_result.end
+      text_index = regex_result.begin
+
+      # puts group_name
+
+      if text_index.nil?
+        raise "Invalid text_index: #{text_index}"
+      end
+      if regex_index.nil?
+        raise "Invalid regex_index: #{regex_index}"
+      end
+
+      offset = raw_match.starts_with?(',') ? 1 : 0
+
+      opening_bracket_index = (text_index + group_name.size) + offset
+      closing_bracket_index = bracket_pairs[opening_bracket_index]
+      content_start = opening_bracket_index + 1
+
+      content = fields_text[content_start...closing_bracket_index]
+      content = remove_nest_groups(content)
+
+      while nest_stack.size > 0 && closing_bracket_index > nest_stack[nest_stack.size - 1][:closing_bracket_index]
+        if nest_stack.size
+          nest_stack.pop
+        end
+      end
+
+      nest_stack.push({
+        group_name:            group_name,
+        closing_bracket_index: closing_bracket_index,
+      })
+
+      if content
+        properties = content.split(',')
+
+        properties.each do |prop|
+          nest_list = nest_stack.map { |nest_prop| nest_prop[:group_name] }
+
+          if prop.includes?('/')
+            nest_list = nest_list.concat(prop.split('/'))
+          else
+            nest_list.push(prop)
+          end
+
+          property_value = get_hash_property(hash, nest_list)
+          puts property_value
+          set_hash_property(final_hash, nest_list, property_value)
+        end
+      end
+    end
+
+    final_hash
+  end
+
+  def self.remove_nest_groups(text : String)
+    content_bracket_pairs = get_bracket_pairs(text, false)
+
+    content_bracket_pairs.each_key.to_a.reverse.each do |opening_bracket|
+      closing_bracket = content_bracket_pairs[opening_bracket]
+      last_comma = last_index_of(text, ',', opening_bracket)
+
+      text = text[0...last_comma] + text[closing_bracket + 1...text.size]
+    end
+
+    return text.starts_with?(',') ? text[1...text.size] : text
+  end
+
+  def self.get_bracket_pairs(text : String, recursive = true)
+    istart = [] of Int64
+    d = {} of Int64 => Int64
+
+    text.each_char_with_index do |char, index|
+      if char == '('
+        istart.push(index.to_i64)
+      end
+
+      if char == ')'
+        begin
+          opening = istart.pop
+          if recursive || (!recursive && istart.size == 0)
+            d[opening] = index.to_i64
+          end
+        rescue e : Exception
+          puts "Too many closing parentheses"
```

Comment:
Should this be `raise "Too many closing parentheses"`?. `fields=usage(users))` returns 
```json
{"error": "Invalid property: ')'"}
```

Reply:
Raising seems reasonable.
Fixed with https://github.com/afrmtbl/invidious/commit/9a5b52244d6a7cf8533d2813c71b5994328282fe